File: backend/src/modules/staff/models/Staff.ts
import mongoose from "mongoose";

const staffSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    email: { type: String, required: true },
    password: { type: String, select: false },
    phone: { type: String, required: true },
    position: { type: String, required: true }, // e.g., "Teacher", "Principal", "HOD", "Librarian"
    department: String, // e.g., "Science", "Arts", "Math"
    qualification: String,
    address: String,
    dateOfBirth: String,
    gender: { type: String, enum: ["Male", "Female", "Other"] },
    joinDate: String,
    status: { type: String, enum: ["Active", "Inactive", "On Leave"], default: "Active" },
    bankName: String,
    accountNumber: String,
    ifscCode: String,
    accountHolderName: String,
    panNumber: String,         // PAN (Permanent Account Number)
    ossId: String,             // CBSE OSS ID (teachers)
    workHistoryDoc: String,   // base64 — past working history document
    offerLetterDoc: String,   // base64 — previous school offer letter (teachers)
    identityDoc: String,      // base64 — Aadhaar / identity proof
    salaryStructureId: { type: mongoose.Schema.Types.ObjectId, ref: "SalaryStructure" }, // Reference to salary structure
    schoolId: { type: mongoose.Schema.Types.ObjectId, ref: "School", required: true },
  },
  { timestamps: true }
);

export default mongoose.model("Staff", staffSchema);

File: go-backend/internal/modules/staff/domain/staff.go
package domain

import "time"

type Staff struct {
	ID             string    `json:"_id"`
	SchoolID       string    `json:"schoolId"`
	Name           string    `json:"name"`
	Email          string    `json:"email"`
	Password       string    `json:"password,omitempty"`
	Phone          string    `json:"phone"`
	Position       string    `json:"position"`
	Department     string    `json:"department"`
	Qualification  string    `json:"qualification"`
	Address        string    `json:"address"`
	DateOfBirth    string    `json:"dateOfBirth"`
	Gender         string    `json:"gender"`
	JoinDate       string    `json:"joinDate"`
	Status         string    `json:"status"`
	BankName       string    `json:"bankName"`
	AccountNumber  string    `json:"accountNumber"`
	IFSCCode       string    `json:"ifscCode"`
	AccountHolder  string    `json:"accountHolderName"`
	PANNumber      string    `json:"panNumber"`
	SalaryStructID string    `json:"salaryStructureId"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

type LeaveApplication struct {
	ID          string    `json:"_id"`
	SchoolID    string    `json:"schoolId"`
	TeacherID   string    `json:"teacherId"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	LeaveType   string    `json:"leaveType"`
	Status      string    `json:"status"`
	FileName    string    `json:"fileName"`
	FileData    string    `json:"fileData"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}
